games-roguelike/nethack/nethack-3.6.1.ebuild

```diff
@@ -0,0 +1,128 @@
+
+	use X && append-cflags -DX11_GRAPHICS -DUSE_XPM
+
+	LOCAL_MAKEOPTS=(
+		CC="$(tc-getCC)" CFLAGS="${CFLAGS}" LFLAGS="${LDFLAGS}"
+		WINTTYLIB="$($(tc-getPKG_CONFIG) --libs ncurses)"
+		HACKDIR="${EPREFIX}/usr/$(get_libdir)/nethack" INSTDIR="${ED}/usr/$(get_libdir)/nethack"
+		SHELLDIR="${ED}/usr/bin" VARDIR="${ED}/var/games/nethack"
+		)
+
+	emake "${LOCAL_MAKEOPTS[@]}" nethack recover Guidebook spec_levs
+
+	# Upstream still has some parallel compilation bugs
+	emake -j1 "${LOCAL_MAKEOPTS[@]}" all
+}
+
+src_install() {
+	emake "${LOCAL_MAKEOPTS[@]}" install
+
+	mv "${ED}/usr/$(get_libdir)/nethack/recover" "${ED}/usr/bin/recover-nethack" || die "Failed to move recover-nethack"
+
+	doman doc/nethack.6
+	newman doc/recover.6 recover-nethack.6
+	dodoc doc/Guidebook.txt
+
+	insinto /etc
+	newins sys/unix/sysconf nethack.sysconf
+
+	insinto /etc/skel
+	newins "${FILESDIR}/${PN}-3.6.0-nethackrc" .nethackrc
+
+	if use X ; then
+		cd "${S}/win/X11" || die "Failed to enter win/X11 directory"
+
+		mkdir -p "${ED}/etc/X11/app-defaults/" || die "Failed to make app-defaults directory"
+		mv "${ED}/usr/$(get_libdir)/nethack/NetHack.ad" "${ED}/etc/X11/app-defaults/" || die "Failed to move NetHack.ad"
+
+		newicon nh_icon.xpm nethack.xpm
+		make_desktop_entry ${PN} Nethack
+
+		# install nethack fonts
+		bdftopcf -o nh10.pcf nh10.bdf || die "Converting fonts failed"
+		bdftopcf -o ibm.pcf ibm.bdf || die "Converting fonts failed"
+		insinto "/usr/$(get_libdir)/nethack/fonts"
+		doins *.pcf
+		mkfontdir "${ED}/usr/$(get_libdir)/nethack/fonts" || die "mkfontdir failed"
+	fi
+
+	rm -r "${ED}/var/games/nethack" || die "Failed to clean var/games/nethack"
+	keepdir /var/games/nethack/save
+}
+
+pkg_preinst() {
+	enewgroup gamestat 36
+
+	fowners root:gamestat /var/games/nethack /var/games/nethack/save
+	fperms 2770 /var/games/nethack /var/games/nethack/save
+
+	fowners root:gamestat "/usr/$(get_libdir)/nethack/nethack"
+	fperms g+s "/usr/$(get_libdir)/nethack/nethack"
+}
+
+pkg_postinst() {
+	cd "${EROOT}/var/games/nethack" || die "Failed to enter ${EROOT}/var/games/nethack directory"
+
+	# Transition mechanism for <nethack-3.6.1 ebuilds. It's perfectly safe, so we'll just run it unconditionally.
+	chmod 2770 . save || die "Failed to chmod statedir"
+
+	# Those files can't be created earlier because we don't want portage to wipe them during upgrades
+	( umask 007 && touch logfile perm record xlogfile ) || die "Failed to create log files"
+
+	# Instead of using a proper version header in its save files, nethack checks for incompatibilities
+	# by comparing the mtimes of save files and its own binary. This would require admin interaction even
+	# during upgrades which don't change the file format, so we'll just touch the files and warn the admin
+	# manually in case of compatibility issues.
+	( shopt -s nullglob
+		local saves=( bones* save/* )
+		test ${#saves[@]} -gt 0 && touch -c "${saves[@]}"
```

Comment:
Nonono, no `test` in bash, use `[[ ... ]]`.

Also you can make it simpler: `[[ ${saves[@]} ]]`, or optionally with `-n`.

Reply:
Sorry, my portability instincts took over. :)